packages/zoltan2/src/algorithms/partition/Zoltan2_AlgMultiJagged.hpp

```diff
@@ -6559,18 +6521,22 @@ void AlgMJ<mj_scalar_t, mj_lno_t, mj_gno_t, mj_part_t, mj_node_t>::
     //migrate coordinates
     {
       // coordinates in MJ are LayoutLeft since Tpetra Multivector gives LayoutLeft
-      typename decltype(this->mj_coordinates)::HostMirror host_src_coordinates =
-        Kokkos::create_mirror_view(this->mj_coordinates);
+      auto host_src_coordinates = Kokkos::create_mirror_view(
+        Kokkos::Serial(), this->mj_coordinates);
       Kokkos::deep_copy(host_src_coordinates, this->mj_coordinates);
       Kokkos::View<mj_scalar_t**, Kokkos::LayoutLeft, device_t>
         dst_coordinates(Kokkos::ViewAllocateWithoutInitializing("mj_coordinates"),
         num_incoming_gnos, this->coord_dim);
-      typename decltype(dst_coordinates)::HostMirror
-        host_dst_coordinates = Kokkos::create_mirror_view(dst_coordinates);
+      auto host_dst_coordinates = Kokkos::create_mirror_view(
+        Kokkos::HostSpace(), dst_coordinates);
       for(int i = 0; i < this->coord_dim; ++i) {
-        Kokkos::View<mj_scalar_t *, device_t> sub_host_src_coordinates
-          = Kokkos::subview(host_src_coordinates, Kokkos::ALL, i);
-        Kokkos::View<mj_scalar_t *, device_t> sub_host_dst_coordinates
+        Kokkos::View<mj_scalar_t*, Kokkos::Serial> sub_host_src_coordinates;
+        // view could be size 0 if graph was not distributed
+        if(host_src_coordinates.extent(0) != 0) {
```

Comment:
Is this check for extent != 0 needed?  There is not a similar check of the extent for sub_host_src_weights or below.

Reply:
@kddevin I forgot I had included that change which was specific just to a chaco test. This change is in the #ifdef block for using Zoltan to migrate instead of Tpetra, which is off unless someone does a manual code change. So it's not currently used (or tested). When I did the original PR updating MJ to kokkos that code path was actually left broken for the chaco test and I didn't try it until later. So this was to fix that unused path which I have been maintaining. However if we want to keep that code I should probably add some testing for it.  I expect no one is using the Zoltan option as it was in a broken state for many weeks without any complaints.

I will review the situation to better answer your question. If you agree I could address further issues with this as a follow up PR since I don't think anyone is using that code right now. So it should be safe to allow the change. There is a similar change for the Tpetra path which has been in for a while.